### ISSUE TITLE
.bower.json contains illegal json (sometimes)

### ISSUE THREAD

**Issue by joelharkes** [bower/bower#2067]
installing package with bower i get 2 bower.json files: bower.json and .bower.json.

the .bower.json file contains:

``` javascript
{
  "name": "jquery.serialscroll",
  "version": "1.3.1",
  "description": "Animated scrolling of series with jQuery",
  "homepage": "https://github.com/flesler/jquery.serialScroll",
  "main": [
    "./jquery.serialScroll.js",
    "./jquery.serialScroll.min.js"
  ],
  "dependencies": {
    "jquery": ">=1.8",
    "jquery.scrollTo": ">=2.1.0"
  },
  "keywords": [
    "slideshow",
    "sequence",
    "animated",
    "animation",
    "scrolling",
    "scroll",
    "prev",
    "next"
  ],
  "author": {
    "name": "Ariel Flesler",
    "web": "http://flesler.blogspot.com/"
  },
  "_release": "1.3.1",
  "_resolution": {
    "type": "version",
    "tag": "1.3.1",
    "commit": "eb3c573bee7f0fa60c46c31822f03c2479eb6b6e"
  },
  "_source": "git://github.com/flesler/jquery.serialScroll.git",
  "_target": "*",
  "_originalSource": "jquery.serialScroll"
}ll"
}
```

done with command install bower

and in json file:

``` javscript
"ignore": [
 "**/.*",
    "node_modules",
    "bower_components"
  ],
  "dependencies": {
    "bootstrap": ">=3.3.5",
    "bootstrap-touchspin": "~3.0.3",
    "font-awesome": "~4.4.0",
    "javascript-equal-height-responsive-rows": "*",
    "jquery": ">=2.1.4",
    "jquery.scrollTo": "~2.1.2",
    "jquery.serialscroll": "jquery.serialScroll#~1.3.1",
    "jquery-mousewheel": "~3.1.13",
    "jquery-parallax": "~1.1.3",
    "jquery-ui": "~1.11.4",
    "malihu-custom-scrollbar-plugin": "~3.1.3",
    "minimalect": "*",
    "moment": "~2.10.6",
    "imagesloaded": "~3.2.0"
  },
  "resolutions": {
    "jquery.scrollTo": "~2.1.2",
    "jquery": ">=1.9.0"
  }
```

**Comment by joelharkes** [comment]
I can reproduce it by runing bower install twice after eachother

```
cmd>bower install
bower                       EMALFORMED Failed to read C:\Git\nextgen-cartwise\Frontend\Source\Frontend\skin\bower_components\jquery.serialscroll\.bower.json

Additional error details:
Unexpected token l

```

**Comment by pertrai1** [comment]
@joelharkes Just out of curiosity and for learning, in the json file, does 
`jquery.serialscroll": "jquery.serialScroll#~1.3.1"` 
have to include the name of the package, or can it just be `~1.3.1` ?

**Comment by pertrai1** [comment]
@joelharkes - just wanted to follow up about this issue and if you have been able to resolve?

**Comment by joelharkes** [comment]
Yes i'm pretty use its not an issue of this plugin, i haven't had time to test just using `~1.3.1`. il do that as soon as possible.

**Comment by KonradKiss** [comment]
We're seeing this issue as well. The error message in the console after a prune and install: 

```bash

konrad-mbp:X konrad$ bower install
...
bower install               jQuery#2.1.4
bower error                 Unexpected string

Stack trace:
SyntaxError: Unexpected string
    at Object.parse (native)
    at /usr/local/lib/node_modules/bower/lib/core/Manager.js:208:37
    at _fulfilled (/usr/local/lib/node_modules/bower/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/bower/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/bower/node_modules/q/q.js:796:13)
    at /usr/local/lib/node_modules/bower/node_modules/q/q.js:604:44
    at runSingle (/usr/local/lib/node_modules/bower/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/bower/node_modules/q/q.js:125:13)
    at process._tickCallback (node.js:355:11)

Console trace:
Error
    at StandardRenderer.error (/usr/local/lib/node_modules/bower/lib/renderers/StandardRenderer.js:83:37)
    at Logger.<anonymous> (/usr/local/lib/node_modules/bower/bin/bower:116:30)
    at Logger.emit (events.js:107:17)
    at Logger.emit (/usr/local/lib/node_modules/bower/node_modules/bower-logger/lib/Logger.js:29:39)
    at /usr/local/lib/node_modules/bower/lib/commands/index.js:48:20
    at _rejected (/usr/local/lib/node_modules/bower/node_modules/q/q.js:844:24)
    at /usr/local/lib/node_modules/bower/node_modules/q/q.js:870:30
    at Promise.when (/usr/local/lib/node_modules/bower/node_modules/q/q.js:1122:31)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/bower/node_modules/q/q.js:788:41)
    at /usr/local/lib/node_modules/bower/node_modules/q/q.js:604:44
System info:
Bower version: 1.7.2
Node version: 0.12.7
OS: Darwin 15.2.0 x64

konrad-mbp:X konrad$ bower install

bower EMALFORMED    Failed to read /Users/konrad/Dev/X/client/bower_components/jquery/.bower.json

Additional error details:
Unexpected string
```

The first install or update renders the `.bower.json` file, this is the result in this case:

```json
{
  "name": "jquery",
  "version": "2.1.4",
  "main": "dist/jquery.js",
  "license": "MIT",
  "ignore": [
    "**/.*",
    "build",
    "dist/cdn",
    "speed",
    "test",
    "*.md",
    "AUTHORS.txt",
    "Gruntfile.js",
    "package.json"
  ],
  "devDependencies": {
    "sizzle": "2.1.1-jquery.2.1.2",
    "requirejs": "2.1.10",
    "qunit": "1.14.0",
    "sinon": "1.8.1"
  },
  "keywords": [
    "jquery",
    "javascript",
    "library"
  ],
  "homepage": "https://github.com/jquery/jquery",
  "_release": "2.1.4",
  "_resolution": {
    "type": "version",
    "tag": "2.1.4",
    "commit": "7751e69b615c6eca6f783a81e292a55725af6b85"
  },
  "_source": "git://github.com/jquery/jquery.git",
  "_target": "~2.1.4"
}">=1.6.3"
}
```

That line before the last is the culprit. Not sure where that 1.6.3 is coming from (perhaps .cache), jquery 2.1.4 is resolved in the install process (our project's json sets ~2.1.4 up as a dependency). To be clear, I don't think the problem has anything to do with jQuery at all.

I made sure bower.json does not contain anything other than ascii. It's utf-8 encoded and has LF linebreaks, just in case..

Fixing the `.bower.json` file and running `bower install` again gives me this:

```bash
bower install               jquery#2.1.4
bower skipped               jquery was not installed because there is already a non-bower directory with that name in the components directory (client/bower_components/jquery). You can force installation with --force.
bower install               jQuery#2.1.4

jquery#2.1.4 ../../.cache/bower/packages/8ccd8ff8cca80bc6237ebfb752ea634b/2.1.4

jQuery#2.1.4 client/bower_components/jQuery
```

Not sure if any of this helps, unfortunately we can't reliably reproduce the issue with anyone else in my team.

Also, really not sure if this doesn't just complicate things, but one other developer saw this only once and realized that the repo of jquery changed to the old repo instead of the new -dist repo while this error was present.

**Comment by BenMann** [comment]
Can you share the bower.json from which you get that result — the one from the package that you run `bower install` on? I'm not sure i understand which steps you take to get this, and which `bower.json` is flawed in the end. Can you list all the steps?

**Comment by KonradKiss** [comment]
Sure, here you go:

```json
{
  "name": "px",
  "version": "0.0.1",
  "dependencies": {
    "angular": ">=1.4.*",
    "angular-animate": "~1.4.3",
    "angular-bootstrap": "~0.14.*",
    "angular-busy": "~4.1.3",
    "angular-cookies": ">=1.4.*",
    "angular-formly-templates-ennosol": "git+https://github.com/ennosol/angular-formly-templates-ennosol.git#master",
    "angular-leaflet-directive": "~0.9.*",
    "angular-loading-bar": "~0.8.0",
    "angular-moment": "~1.0.0",
    "angular-resource": ">=1.4.*",
    "angular-sanitize": ">=1.4.*",
    "angular-scroll": "~0.7.1",
    "angular-translate": "~2.7.2",
    "angular-translate-loader-static-files": "~2.7.2",
    "angular-translate-storage-cookie": "~2.7.2",
    "angular-translate-storage-local": "~2.7.2",
    "angular-ui-router": "~0.2.15",
    "angular-ui-switch": "~0.1.0",
    "angular-sortable-view": "git+https://github.com/kamilkp/angular-sortable-view#~0.0.13",
    "bootstrap": "=3.3.5",
    "components-modernizr": "~2.8.3",
    "es5-shim": "~3.0.1",
    "fastclick": "~1.0.6",
    "font-awesome": ">=4.1.0",
    "jquery": "~2.1.4",
    "json3": "~3.3.1",
    "lodash": "~2.4.1",
    "moment": "~2.10.6",
    "ngstorage": "~0.3.7",
    "ngSweetAlert": "angular-sweetalert#~1.1.0",
    "themify-icons": "~0.1.0",
    "toastr": "~2.1.2",
    "leaflet.awesome-markers": "git+https://github.com/ennosol/Leaflet.awesome-markers.git#2.0/develop",
    "leaflet.polyline-snakeanim": "git+https://github.com/IvanSanchez/Leaflet.Polyline.SnakeAnim#~0.1.0",
    "leaflet.encoded": "git+https://github.com/jieter/Leaflet.encoded#~0.0.7",
    "jquery-ui": "jquery.ui#~1.11.4",
    "leaflet": "~1.0",
    "Leaflet.markercluster": "Leaflet/Leaflet.markercluster#1.0.0-beta.2.0",
    "Leaflet.fullscreen": "Leaflet/Leaflet.fullscreen#~0.0.4",
    "Leaflet.label": "Leaflet/Leaflet.label#leaflet-master",
    "angular-visibility-change": "~0.1.0",
    "Leaflet.EasyButton": "~1.1.2",
    "Leaflet.singleclick": "~0.4.0",
    "leaflet-icon-pulse": "mapshakers/leaflet-icon-pulse#~0.1.0",
    "angular-simple-logger": "~0.1.4",
    "sprintf": "~1.0.3",
    "Leaflet.hotline": "iosphere/Leaflet.hotline#~0.2.0",
    "zoomcharts": "git+https://github.com/ennosol/zoomcharts#1.11.4",
    "perfect-scrollbar": "~0.6.1",
    "angular-sockjs": "~0.1.0",
    "sockjs": "sockjs/sockjs-client#~1.0.3",
    "bootstrap-datepicker": "~1.5.0"
  },
  "devDependencies": {
    "angular-mocks": ">=1.4.*",
    "angular-scenario": ">=1.4.*",
    "bardjs": "~0.1.4"
  },
  "overrides": {
    "zoomcharts": {
      "main": [
        "lib/zoomcharts.js"
      ]
    },
    "select2": {
      "main": [
        "dist/css/select2.css",
        "dist/js/select2.full.js"
      ]
    },
    "bootstrap-touchspin": {
      "main": [
        "dist/jquery.bootstrap-touchspin.css",
        "dist/jquery.bootstrap-touchspin.js"
      ]
    },
    "bootstrap": {
      "main": [
        "dist/css/bootstrap.css",
        "dist/js/bootstrap.js"
      ]
    },
    "Leaflet.fullscreen": {
      "main": [
        "dist/leaflet.fullscreen.css",
        "dist/Leaflet.fullscreen.js"
      ]
    },
    "leaflet.awesome-markers": {
      "main": [
        "dist/leaflet.awesome-markers.css",
        "dist/leaflet.awesome-markers.js",
        "images/markers-soft.png"
      ]
    },
    "leaflet-icon-pulse": {
      "main": [
        "src/L.Icon.Pulse.js",
        "src/L.Icon.Pulse.css"
      ]
    },
    "font-awesome": {
      "main": [
        "css/font-awesome.min.css"
      ]
    },
    "jquery-ui": {
      "main": [
        "themes/base/core.css",
        "themes/base/slider.css",
        "themes/base/theme.css",
        "ui/core.js",
        "ui/widget.js",
        "ui/mouse.js",
        "ui/slider.js"
      ]
    },
    "angular-sortable-view": {
      "main": [
        "src/angular-sortable-view.js"
      ]
    },
    "sprintf": {
      "main": [
        "dist/sprintf.min.js",
        "dist/angular-sprintf.min.js"
      ]
    },
    "Leaflet.hotline": {
      "main": [
        "dist/leaflet.hotline.js"
      ]
    },
    "moment": {
      "main": [
        "moment.js",
        "locale/hu.js"
      ]
    },
    "angular-cron-jobs": {
      "main": [
        "dist/angular-cron-jobs.js",
        "dist/angular-cron-jobs.min.css"
      ]
    }
  },
  "resolutions": {
    "angular": ">= 1.4.5",
    "angular-translate": "~2.7.2",
    "angular-translate-storage-cookie": "~2.7.2",
    "angular-cookies": ">=1.4.*",
    "moment": "~2.10.6",
    "fastclick": "~1.0.6",
    "angular-formly": "~6.23.2",
    "select2": "~4.0.0",
    "angular-bootstrap": "~0.13.*",
    "angular-animate": "~1.4.3",
    "leaflet": "~1.0"
  }
}
```

These are the preinstall build steps in package.json: 
`npm install -g bower grunt-cli && npm run vendor-clean && bower update -f`

vendor-clean is `npm prune && bower prune`

The flawed `.bower.json` appears in the bower_components/jquery directory (starts with a dot, I imagine this is a build cache).

**Comment by BenMann** [comment]
Thanks! :trophy:   
I can replicate this, even without 
`npm install -g bower grunt-cli && npm run vendor-clean && bower update -f` and
`npm prune && bower prune`   
I'm a little disturbed that on first sight jquery is the only package causing this (having this issue). I'll try to reduce this and figure out what causes it.. meanwhile.. if you install jquery directly you get this (`.bower.json`):
```
{
  "name": "jquery",
  "version": "2.1.4",
  "main": "dist/jquery.js",
  "license": "MIT",
  "ignore": [
    "**/.*",
    "build",
    "dist/cdn",
    "speed",
    "test",
    "*.md",
    "AUTHORS.txt",
    "Gruntfile.js",
    "package.json"
  ],
  "devDependencies": {
    "sizzle": "2.1.1-jquery.2.1.2",
    "requirejs": "2.1.10",
    "qunit": "1.14.0",
    "sinon": "1.8.1"
  },
  "keywords": [
    "jquery",
    "javascript",
    "library"
  ],
  "homepage": "https://github.com/jquery/jquery-dist",
  "_release": "2.1.4",
  "_resolution": {
    "type": "version",
    "tag": "2.1.4",
    "commit": "7751e69b615c6eca6f783a81e292a55725af6b85"
  },
  "_source": "git://github.com/jquery/jquery-dist.git",
  "_target": "~2.1.4",
  "_originalSource": "jquery",
  "_direct": true
}
```

When required as a depency as in your case, this is the result:

```
{
  "name": "jquery",
  "version": "2.1.4",
  "main": "dist/jquery.js",
  "license": "MIT",
  "ignore": [
    "**/.*",
    "build",
    "dist/cdn",
    "speed",
    "test",
    "*.md",
    "AUTHORS.txt",
    "Gruntfile.js",
    "package.json"
  ],
  "devDependencies": {
    "sizzle": "2.1.1-jquery.2.1.2",
    "requirejs": "2.1.10",
    "qunit": "1.14.0",
    "sinon": "1.8.1"
  },
  "keywords": [
    "jquery",
    "javascript",
    "library"
  ],
  "homepage": "https://github.com/jquery/jquery",
  "_release": "2.1.4",
  "_resolution": {
    "type": "version",
    "tag": "2.1.4",
    "commit": "7751e69b615c6eca6f783a81e292a55725af6b85"
  },
  "_source": "git://github.com/jquery/jquery.git",
  "_target": "~2.1.4"
}"~2.1.4"
}
```

Notice how the `-dist` is missing? If we find out why, we can probably fix this...

**Comment by BenMann** [comment]
Here's a super reduced test-case: 
1) `bower cache clean`
2) `bower install` the following:
```
{
  "name": "px",
  "version": "0.0.1",
  "dependencies": {
    "jquery": "~2.1.4",
    "angular-ts-select2": "~1.0.0"
  }
}
```
If you follow the installation of `jquery` and `jQuery` through the output below, you notice that the right `jquery` is installed first, after which bower resolves the dependency (`jQuery`) that is listed in `angular-ts-select2`. They are also installed in that order.

```
bower install
bower not-cached    git://github.com/jquery/jquery-dist.git#~2.1.4
bower resolve       git://github.com/jquery/jquery-dist.git#~2.1.4
bower not-cached    git://github.com/tscheinecker/angular-ts-select2.git#~1.0.0
bower resolve       git://github.com/tscheinecker/angular-ts-select2.git#~1.0.0
bower download      https://github.com/tscheinecker/angular-ts-select2/archive/v1.0.0-alpha.20150603.1.tar.gz
bower download      https://github.com/jquery/jquery-dist/archive/2.1.4.tar.gz
bower extract       angular-ts-select2#~1.0.0 archive.tar.gz
bower resolved      git://github.com/tscheinecker/angular-ts-select2.git#1.0.0-alpha.20150603.1
bower not-cached    git://github.com/ivaynberg/select2.git#~4.0.0
bower resolve       git://github.com/ivaynberg/select2.git#~4.0.0
bower not-cached    git://github.com/jquery/jquery.git#~2.1.4
bower resolve       git://github.com/jquery/jquery.git#~2.1.4
bower not-cached    git://github.com/angular/bower-angular.git#1.2 - 1.4
bower resolve       git://github.com/angular/bower-angular.git#1.2 - 1.4
bower extract       jquery#~2.1.4 archive.tar.gz
bower resolved      git://github.com/jquery/jquery-dist.git#2.1.4
bower download      https://github.com/ivaynberg/select2/archive/4.0.1.tar.gz
bower download      https://github.com/angular/bower-angular/archive/v1.4.8.tar.gz
bower download      https://github.com/jquery/jquery/archive/2.1.4.tar.gz
bower extract       angular#1.2 - 1.4 archive.tar.gz
bower resolved      git://github.com/angular/bower-angular.git#1.4.8
bower extract       select2#~4.0.0 archive.tar.gz
bower extract       jQuery#~2.1.4 archive.tar.gz
bower resolved      git://github.com/jquery/jquery.git#2.1.4
bower invalid-meta  select2 is missing "ignore" entry in bower.json
bower resolved      git://github.com/ivaynberg/select2.git#4.0.1
bower install       angular-ts-select2#1.0.0-alpha.20150603.1
bower install       jquery#2.1.4
bower install       angular#1.4.8
bower install       jQuery#2.1.4
bower install       select2#4.0.1
bower error         Unexpected string

Stack trace:
SyntaxError: Unexpected string
    at Object.parse (native)
    at /usr/local/lib/node_modules/bower/lib/core/Manager.js:208:37
    at _fulfilled (/usr/local/lib/node_modules/bower/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/bower/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/bower/node_modules/q/q.js:796:13)
    at /usr/local/lib/node_modules/bower/node_modules/q/q.js:604:44
    at runSingle (/usr/local/lib/node_modules/bower/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/bower/node_modules/q/q.js:125:13)
    at nextTickCallbackWith0Args (node.js:433:9)
    at process._tickCallback (node.js:362:13)

Console trace:
Error
    at StandardRenderer.error (/usr/local/lib/node_modules/bower/lib/renderers/StandardRenderer.js:83:37)
    at Logger.<anonymous> (/usr/local/lib/node_modules/bower/bin/bower:116:30)
    at emitOne (events.js:78:13)
    at Logger.emit (events.js:170:7)
    at Logger.emit (/usr/local/lib/node_modules/bower/node_modules/bower-logger/lib/Logger.js:29:39)
    at /usr/local/lib/node_modules/bower/lib/commands/index.js:48:20
    at _rejected (/usr/local/lib/node_modules/bower/node_modules/q/q.js:844:24)
    at /usr/local/lib/node_modules/bower/node_modules/q/q.js:870:30
    at Promise.when (/usr/local/lib/node_modules/bower/node_modules/q/q.js:1122:31)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/bower/node_modules/q/q.js:788:41)
System info:
Bower version: 1.7.2
Node version: 5.2.0
OS: Darwin 14.5.0 x64
``` 
We also see that `jQuery` does not resolve to the `jquery-dist` [repo](https://github.com/jquery/jquery-dist), which actually contains the `bower.json` but simply to `jquery/jquery`, where no `bower.json` is present. Even though I'd love to just point the `jQuery` package to the right source in the Bower registry, I think we should try to find out what exactly is causing the invalid `.bower.json` (below) first:
```
...
   "_source": "git://github.com/jquery/jquery.git",
   "_target": "~2.1.4"
 }"~2.1.4"
}
```

**Comment by KonradKiss** [comment]
Great job tracking it this far! Thanks a lot for sharing the simpler repro case. I'll let you know when I find something new.

**Comment by BenMann** [comment]
I'm pretty sure I have just figured out why the first one fails too! it's not `jquery` vs `jQuery` but `jquery.serialscroll` vs. `jquery.serialScroll` in this case. The package `jquery-parallax` is requiring it in camel-case, while the original `bower.json` that we install is using it all lower-case. 
```
[...]
/bower_components/jquery-parallax/bower.json:
    6      "jquery": "~1.8.3",
    7      "jquery.scrollTo": "~1.4.6",
    8:     "jquery.serialScroll": "*"
    9    },
   10    "ignore": [
[...]
```

This is also mirrored during installation (see ">>"):
 
```
bower javascript-equal-height-responsive-rows#*          install javascript-equal-height-responsive-rows#9e01fe3612
bower minimalect#*                                       install minimalect#d9c4d9680f
bower jquery.scrollTo#>=2.1.0                            install jquery.scrollTo#2.1.2
bower bootstrap-touchspin#~3.0.3                         install bootstrap-touchspin#3.0.3
>> bower jquery.serialscroll#~1.3.1                         install jquery.serialscroll#1.3.1
bower jquery#>=1.8                                       install jquery#2.2.0
bower imagesloaded#~3.2.0                                install imagesloaded#3.2.0
bower jquery-mousewheel#~3.1.13                          install jquery-mousewheel#3.1.13
bower eventie#~1.0.4                                     install eventie#1.0.6
bower eventEmitter#>=4.2 <5.0                            install eventEmitter#4.3.0
bower malihu-custom-scrollbar-plugin#~3.1.3              install malihu-custom-scrollbar-plugin#3.1.3
bower moment#~2.10.6                                     install moment#2.10.6
bower jquery-parallax#~1.1.3                             install jquery-parallax#1.1.3
>> bower jquery.serialScroll#*                              install jquery.serialScroll#1.3.1
bower jquery-ui#~1.11.4                                  install jquery-ui#1.11.4
bower font-awesome#~4.4.0                                install font-awesome#4.4.0
bower bootstrap#>=3.3.5                                  install bootstrap#3.3.6
bower                                                      error Unexpected end of input
```
So that solves the question why it's jquery in one case and serialscroll in the other case, which have corrupted `.bower.json`s. One quirky thing remains: `bower lookup jquery.serialscroll` says the package doesn't even exist all lower-case. As a side-note all camel-case packages are not conforming with the spec, but I realize that we needed it [for this](https://github.com/bower/bower/issues/1751).   
The only thing I can think of right now to finally fix this is pointing all camel-case packages to the url of the lower-case packages (ignorantly assuming they are the right ones) or in case no lower-case package exists, make the camel-case lower-case. Thereafter reinstate the rule that only lower-case packages can be registered. I have no idea if this might still break builds for people! :bomb:

**Comment by kmcmichael-twist** [comment]
I'm getting this same error. After bower install, the contents of bower_components/jquery/.bower.json are munged (error on bolded line):

{
  "name": "jQuery",
  "homepage": "https://github.com/jquery/jquery",
  "version": "2.2.0",
  "_release": "2.2.0",
  "_resolution": {
    "type": "version",
    "tag": "2.2.0",
    "commit": "33b548c8e3d43b2ebdfb129fd8086a3b0c905919"
  },
  "_source": "git://github.com/jquery/jquery.git",
  "_target": ">=1.7.0"
**}irejs": "2.1.10",**
    "qunit": "1.14.0",
    "sinon": "1.8.1"
  },
  "keywords": [
    "jquery",
    "javascript",
    "library"
  ],
  "homepage": "https://github.com/jquery/jquery-dist",
  "_release": "2.1.4",
  "_resolution": {
    "type": "version",
    "tag": "2.1.4",
    "commit": "7751e69b615c6eca6f783a81e292a55725af6b85"
  },
  "_source": "git://github.com/jquery/jquery-dist.git",
  "_target": ">= 1.9.0"
}

**Comment by oliviermadre** [comment]
same as @kmcmichael-twist here ! I've got exactly the same fucked up .bower.json file :(

**Comment by terebentina** [comment]
Any news on this? Or workaround/patch? Currently I simply cannot init my project...:(

**Comment by kmcmichael-twist** [comment]
If you manually copy the full module from a separate install into your web
env, your build process *should* skip trying to download the bad module via
bower. That's what we did.

On Tue, Mar 8, 2016 at 12:40 AM, Dan Caragea <notifications@github.com>
wrote:

> Any news on this? Or workaround/patch? Currently I simply cannot init my
> project...:(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bower/bower/issues/2067#issuecomment-193664324>.
>

-- 
|<ipp McMichael
Senior Front End Software Engineer
kmcmichael@twistbioscience.com

**Comment by openfirmware** [comment]
I was running into this issue with a bower.json file and a dependency on `Proj4Leaflet`. Renaming to `proj4leaflet` solved it for me.

**Comment by robsonmafra** [comment]
I had a similar problem with a old project I have. I was getting an error with JQuery and I solved it changing my `bower.json` from `"jQuery": "2.2.4"` to `"jquery": "2.2.4"`.

```
bower jquery#>=1.6.3 <3         install jquery#2.2.4
bower jQuery#2.2.4              install jQuery#2.2.4
bower                             error Unexpected token p in JSON at position 316
EMALFORMED Failed to read \bower_components\jquery\.bower.json
Stack trace:
SyntaxError: Unexpected token p in JSON at position 316
```

**Comment by joelharkes** [comment]
By the looks of the responses, you should check if your dependencies have capitals and remove them!

Im closing my issue, no longer had this problem and bower popularity seems in decline, since webpack2 and browserify makes separation no longer needed.
